inspirehep/modules/workflows/utils.py

```diff
@@ -80,3 +86,57 @@ def log_workflows_action(action, prediction_results,
         }
         audit = WorkflowsAudit(**logging_info)
         audit.save()
+
+
+def with_debug_logging(func):
+    """Generate a debug log with info on what's going to
+    run.
+
+    It tries it's best to use the logging facilities of the object passed or
+    the application context before falling back to the python logging facility.
+
+    """
+    @wraps(func)
+    def _decorator(*args, **kwargs):
+        def _get_obj(args, kwargs):
+            if args:
+                obj = args[0]
+            else:
+                obj = kwargs.get('obj', kwargs.get('record'))
+            return obj
+
+        def _get_logfn(args, kwargs):
+            obj = _get_obj(args, kwargs)
+            if hasattr(obj, 'log') and hasattr(obj.log, 'debug'):
+                logfn = obj.log.debug
+            elif hasattr(current_app, 'log'):
+                logfn = current_app.log.debug
+            else:
+                logfn = LOGGER.debug
+
+            return logfn
+
+        def _proof_run(func, *args, **kwargs):
```

Comment:
Since you have already above `func` (as the actual function that wants to be run) this parameter can be confusing.

You are actually testing here the `logfn` right?